supervision/dataset/classification.py

```diff
@@ -0,0 +1,239 @@
+            output_directory_path (str): The path to the directory where the dataset will be saved.
+
+        Example:
+            ```python
+            >>> import supervision as sv
+
+            >>> cd = sv.ClassificationDataset(...)
+
+            >>> cd.as_multiclass_folder_structure(
+            ...     root_directory_path="./images",
+            ...     output_directory_path="./out",
+            ... )
+        """
+        if not os.path.exists(output_directory_path):
+            os.makedirs(output_directory_path)
+
+        for split in ["train", "test", "valid"]:
+            if not os.path.exists(output_directory_path + "/" + split):
+                os.makedirs(output_directory_path + "/" + split)
+
+        train, test = self.split(split_ratio=0.8)
+        train, val = train.split(split_ratio=0.8)
+
+        for split, dataset in zip(["train", "test", "valid"], [train, test, val]):
+            for class_name in self.classes:
+                if not os.path.exists(
+                    output_directory_path + "/" + split + "/" + class_name
+                ):
+                    os.makedirs(output_directory_path + "/" + split + "/" + class_name)
+
+                for image in dataset.images:
+                    if dataset.annotations[image].class_id[0] == self.classes.index(
+                        class_name
+                    ):
+                        full_dir = os.path.join(root_directory_path, image)
+
+                        cv2.imwrite(
+                            output_directory_path
+                            + "/"
+                            + split
+                            + "/"
+                            + class_name
+                            + "/"
+                            + image,
+                            cv2.imread(full_dir),
+                        )
+
+    @classmethod
+    def from_multiclass_folder_structure(
+        cls, root_directory_path: str
+    ) -> "ClassificationDataset":
+        """
+        Load data from a multiclass folder structure into a ClassificationDataset.
+
+        Args:
+            root_directory_path (str): The path to the dataset directory.
+
+        Returns:
+            ClassificationDataset: The dataset.
+
+        Example:
+            ```python
+            >>> import supervision as sv
+
+            >>> cd = sv.ClassificationDataset.from_multiclass_folder_structure(
+            ...     root_directory_path="./dataset",
+            ... )
+        """
+        classes = os.listdir(root_directory_path + "/train")
+        classes.sort()
+
+        images = {}
+        annotations = {}
+
+        for split in ["train", "test", "val"]:
+            for class_name in classes:
+                for image in os.listdir(
+                    root_directory_path + "/" + split + "/" + class_name
+                ):
+                    images[image] = np.array([[[0, 0, 0]]])
```

Comment:
What is happening here?

Reply:
`images: Dict[str, np.ndarray]` is the data type for `images`. What should be the value in this dictionary?

Reply:
It should be `np.array` - the result of loading the image, at least for now. I plan to introduce lazy loading, but you should load the image here for now.

Reply:
Got it!